cmd/entrypoints/serve.go

```diff
@@ -142,31 +144,11 @@ func newHTTPServer(ctx context.Context, cfg *config.ServerConfig, authContext in
 
 		// This option translates HTTP authorization data (cookies) into a gRPC metadata field
 		gwmuxOptions = append(gwmuxOptions, runtime.WithMetadata(auth.GetHTTPRequestCookieToMetadataHandler(authContext)))
-
-		if cfg.Security.AllowCors {
-			// In addition to serving preflight requests below, the server also needs to send a CORS header for all requests,
-			// including simple requests like GET that do not trigger a preflight call.
-			gwmuxOptions = append(gwmuxOptions, runtime.WithForwardResponseOption(server.GetForwardResponseOptionHandler(
-				cfg.Security.AllowedOrigins)))
-		}
 	}
 
 	// Create the grpc-gateway server with the options specified
 	gwmux := runtime.NewServeMux(gwmuxOptions...)
 
-	// Enable CORS if necessary on the gateway mux by adding a handler for all OPTIONS requests.
-	if cfg.Security.AllowCors {
-		globPattern := server.GetGlobPattern()
-		decorator := auth.GetCorsDecorator(ctx, cfg.Security.AllowedOrigins, cfg.Security.AllowedHeaders)
```

Comment:
how does it handle OPTIONS calls if this bit is removed?

Reply:
The Cors middleware wraps the mux and processes all requests, passing them through to the mux unless there's an error of some sort (disallowed origin, etc). It will handle all OPTIONS requests and add the appropriate headers based on the configuration.